config/deploy/docker/lib/aws_sdk_helpers.rb

```diff
@@ -0,0 +1,112 @@
+    define_singleton_method(:cw) { Aws::CloudWatch::Client.new }
+    define_singleton_method(:cwl) { Aws::CloudWatchLogs::Client.new }
+    define_singleton_method(:ssm) { Aws::SSM::Client.new }
+
+    define_method(:iam)              { AwsSdkHelpers::ClientMethods.iam }
+    define_method(:elbv2)            { AwsSdkHelpers::ClientMethods.elbv2 }
+    define_method(:ecr)              { AwsSdkHelpers::ClientMethods.ecr }
+    define_method(:ecs)              { AwsSdkHelpers::ClientMethods.ecs }
+    define_method(:secretsmanager)   { AwsSdkHelpers::ClientMethods.secretsmanager }
+    define_method(:autoscaling)      { AwsSdkHelpers::ClientMethods.autoscaling }
+    define_method(:ec2)              { AwsSdkHelpers::ClientMethods.ec2 }
+    define_method(:cloudwatchevents) { AwsSdkHelpers::ClientMethods.cloudwatchevents }
+    define_method(:cw)               { AwsSdkHelpers::ClientMethods.cw }
+    define_method(:cwl)              { AwsSdkHelpers::ClientMethods.cwl }
+    define_method(:ssm)              { AwsSdkHelpers::ClientMethods.ssm }
+  end
+
+  module Helpers
+    include AwsSdkHelpers::ClientMethods
+
+    def self.cluster_name
+      ENV.fetch('CLUSTER_NAME', 'openpath')
+    end
+
+    def _cluster_name
+      AwsSdkHelpers::Helpers.cluster_name
+    end
+
+    def self.capacity_providers(cluster)
+      r = {}
+      capacity_provider_names = AwsSdkHelpers::ClientMethods.ecs.describe_clusters(clusters: [cluster]).clusters.first.capacity_providers
+      AwsSdkHelpers::ClientMethods.ecs.describe_capacity_providers(capacity_providers: capacity_provider_names).capacity_providers.each do |capacity_provider|
+        asg_name = capacity_provider.auto_scaling_group_provider.auto_scaling_group_arn.split('/').last
+        asg = AwsSdkHelpers::ClientMethods.autoscaling.describe_auto_scaling_groups(auto_scaling_group_names: [asg_name]).auto_scaling_groups.first
+
+        launch_template_id = asg.mixed_instances_policy.launch_template.launch_template_specification.launch_template_id
+        launch_template_versions = AwsSdkHelpers::ClientMethods.ec2.describe_launch_template_versions(launch_template_id: launch_template_id)
+
+        ami_id = launch_template_versions.launch_template_versions[0].launch_template_data.image_id
+
+        r[capacity_provider.name] = {
+          name: capacity_provider.name,
+          ami_id: ami_id,
+        }
+      end
+      return r
+    end
+
+    def _capacity_providers(cluster = nil)
+      cluster ||= _cluster_name
+      @capacity_providers ||= AwsSdkHelpers::Helpers.capacity_providers(cluster)
+    end
+
+    def self.get_capacity_provider_name(namespace = '', which = 'Spot')
+      default_path = "/OpenPath/CapacityProviders/#{which}"
+      namespaced_path = "/#{namespace}/CapacityProviders/#{which}"
+
+      params = AwsSdkHelpers::ClientMethods.ssm.get_parameters(
+        {
+          names: [
+            default_path,
+            namespaced_path,
```

Comment:
Could you comment this method with how this defaulting logic works? Is this related to the transition and will become defunct afterwards? If that's the case, we might want to make a PT to clean this up afterwards.

Reply:
For some added context:
- I'm taking this out because we don't actually need it right now.
- a more flexible, robust, and well-commented version of this functionality will be needed in ecs-deploy-public v2 (aka ecs-deploy-private), but we'll get to that later.